components/browser/engine-gecko-beta/src/main/java/mozilla/components/browser/engine/gecko/GeckoEngine.kt

```diff
@@ -158,6 +160,11 @@ class GeckoEngine(
         this.webExtensionDelegate = webExtensionDelegate
 
         val tabsDelegate = object : WebExtensionController.TabDelegate {
+            // We use this map to:
+            // - Ensures a web extension can only close tabs it has opened
+            // - Find the engine session of a given gecko session
+            private val tabs = WeakHashMap<GeckoEngineSession, String>()
```

Comment:
If we really need this mapping .. would it be an option to put it into our state (e.g. the extension id in the tab) instead of having a separate mini state here?